### PR TITLE
[parametric] Update GRPC headers to list of string tuples

### PR DESCRIPTION
**Pull request by zacharycmontoya**
## Description

Updates the headers in the GRPC protocol to be a list of string tuples. This simulates HTTP traffic where a header may be set multiple times and is a valid test case for the W3C Tracecontext tests. 

This required an update to the GRPC server for each tracing library, which is included in this PR. This should result in passing tests for all languages, but my changes may not be up to the quality expected for other languages. Feel free to commit changes to improve the conventions/approach if desired!

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

File: parametric/apps/nodejs/servicer.js [hunk 1]
```diff
@@ -26,11 +26,11 @@ class Servicer {
             parent.origin = request.origin;
         }
 
-        const { http_headers = {} } = request.http_headers || {};
+        const http_headers = request.http_headers.http_headers || [];
```
Comment (Qard) [inline parametric/apps/nodejs/servicer.js:29]: IIRC there were case where the first level of http_headers would be undefined which was why it was done with the destructuring previously. Might need to check if this still works. 🤔

File: parametric/apps/nodejs/servicer.js [hunk 2]
```diff
@@ -26,11 +26,11 @@ class Servicer {
...
         // Node.js HTTP headers are automatically lower-cased, simulate that here.
         const convertedHeaders = {};
-        for (const [key, value] of Object.entries(http_headers)) {
-            convertedHeaders[key.toLowerCase()] = value;
+        for (const header_tuple of http_headers) {
+            convertedHeaders[header_tuple.key.toLowerCase()] = header_tuple.value;
```
Comment (Qard) [inline parametric/apps/nodejs/servicer.js:33]:
```suggestion
        for (const { key, value } of http_headers) {
            convertedHeaders[key.toLowerCase()] = value;
```

File: parametric/apps/nodejs/servicer.js [hunk 3]
```diff
@@ -58,9 +58,13 @@ class Servicer {
...
+        const http_headers = [];
 
-        tracer.inject(span, 'http_headers', http_headers);
+        tracer.inject(span, 'http_headers', http_headersDict);
+        for (const [key, value] of Object.entries(http_headersDict)) {
+            http_headers.push({key: key, value: value});
+        }
 
         const ctx = span._spanContext
         console.log('InjectHeaders', http_headers, ctx.toTraceparent(), ctx._tags)
```
Comment (Qard) [inline parametric/apps/nodejs/servicer.js:70]: Forgot this console.log was there. Could delete this while you're in there if you want. Not necessary anymore. 😅

**Review by Qard — COMMENTED** [review]
Left a few notes but otherwise LGTM.

File: parametric/apps/nodejs/servicer.js (hunk 1 above)
Comment (zacharycmontoya) [reply parametric/apps/nodejs/servicer.js:29]: With the existing code, I tested a case where I passed `None` into one of the python methods that expected an array, and it seems to work okay 🤷🏼 Maybe we can leave as-is for now?

File: parametric/apps/nodejs/servicer.js (hunk 3 above)
Comment (zacharycmontoya) [reply parametric/apps/nodejs/servicer.js]: Removed both console.log statements in f7f2b19244143ecb77c63ac1fa090bda70492387 😄

File: parametric/apps/nodejs/servicer.js (hunk 2 above)
Comment (zacharycmontoya) [reply parametric/apps/nodejs/servicer.js]: Nice destructuring, thanks for the suggestion! Done in f7f2b19244143ecb77c63ac1fa090bda70492387

**Review by Qard — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]
👍🏻

File: parametric/apps/golang/main.go [hunk 4]
```diff
@@ -66,9 +75,13 @@ func (s *apmClientServer) InjectHeaders(ctx context.Context, args *InjectHeaders
...
 	}
-	distr := map[string]string{}
+	distr := []*HeaderTuple{}
 	for k, v := range headers {
-		distr[k] = v
+		var headerTuple = &HeaderTuple{
+			Key: k,
+			Value: v,
+		}
+		distr = append(distr, headerTuple)
```
Comment (dianashevchenko) [inline parametric/apps/golang/main.go:84]:
```suggestion
		distr = append(distr, &HeaderTuple{Key: k, Value: v})
```

Comment (zacharycmontoya) [reply parametric/apps/golang/main.go]: Fixed in b843e78cfd5be0963f4f60c1c7af5ac01f9360fc

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**